### ISSUE TITLE
Finish new filtering system basics

### ISSUE THREAD

**Issue by blay09** [Dynious/RefinedRelocation#300]
- [x] Add left/top margin to text in multi-line textbox
- [x] Fix whole-word-deletion in multi-line textbox
- [x] Add vertical and horizontal scrolling to multi-line textbox
- [x] Allow separation by both comma and newline in User filter
- [x] Add icons for Creative, Preset and User filter
- [x] Add header to Creative and Preset filter
- [x] Make the top of the filtering UI look good again
- [x] Apply new system to Adv. Filtered GUI as well
- [x] Localization
- [x] Implement the new icons for the filters
- [x] Advanced Filtered Block Extender in GUI -> Adv. Filtered Block Extender (for the looks)
- [x] Turn Redstone Signal Transmission into the other style button
- [x] Add the grey background box around the side thingy in Adv. Filtered Block Extender again
- [x] Add icons to the buttons on empty filter page
- [x] Make things more API-friendly
- [x] Separate options tab (at the top?)
- [x] Icon for the options tab
- [x] Tab button is one pixel off to the left when inactive
- [x] "Remove Filter" button
- [x] Increase height of filter list graphic
- [x] Fix scrollbar not stopping to scroll
- [x] Add Icon for empty filter tab
- [x] Stop multiline textbox from going to start when pressing down at end
- [x] Add tooltip to delete button
- [x] Add tooltips to filter type buttons
- [x] Add help text to the big empty space on the filter settings tab
- [x] Flickering when toggling redstone status button
- [x] GUI goes to first tab when adding a filter (but only once, afterwards it works fine)
- [x] Hide delete button on empty filter tabs
- [x] Help button has no tooltip anymore on user filter
- [x] Remove survival inventory and Search tabs from Creative Filter
- [x] Remove redstone button from WBE

**Comment by Dynious** [comment]
I really like the new Filtering, it's a lot more clean and will be less confusing for the user. This interface will also be more easy for us to update, as the back-end is more sensible than it was :P

There are a few things that might need some improvement though:
1. The filtering system should be more extensible. Preferably no integer ids. This will make it a lot more easy for us to add new filter types. Adding new filter types should preferably also be possible using the API. This was already done in the newFiltering branch.
2. Move block dependent settings like redstone, the advanced options and whitelisting to a separate 'options' tab. This will make the filter interfaces even more clean.
3. Maybe add a more descriptive explanation on how to use the interface. (Add filters by using a button, white/blacklisting can be set by pressing the button etc.)

**Comment by blay09** [comment]
Alright, got all of my points and the ones that you mentioned done now.

Just this GUI is still ugly and I'm not sure what to put there to get rid of the insane emptiness:

![ugly](https://cloud.githubusercontent.com/assets/1933180/7997405/de304d26-0b2c-11e5-8881-35fc90858387.png)

**Comment by blay09** [comment]
![2015-06-05_16 32 42](https://cloud.githubusercontent.com/assets/1933180/8008141/8eed852a-0ba0-11e5-9340-2a8f55b92de0.png)

**Comment by blay09** [comment]
@Dynious Have a look at it please and let me know when I can merge. I tested the conversion from old to new and it worked fine, so existing worlds will continue to work.

There were API changes though!

**Comment by Dynious** [comment]
Looks awesome! To make mods that support the old system work with the new one, we only need to change back the names to the old ones. (No mod has used the IFilterGUI methods). We could deprecate them and make interfaces with the new names that just extend the old ones until 1.8 and then we move functionality to the new interfaces.

**Comment by blay09** [comment]
Okay, I'll do that then!

**Comment by blay09** [comment]
Tested it with Compact Drawers and Enhanced Inventories and it works perfectly fine now.

Anything else left to do before I can merge it into master?

**Comment by Dynious** [comment]
Nope! Let's do it :D
